gnd/src/main/java/com/google/android/gnd/persistence/sync/TileDownloadWorkManager.java

```diff
@@ -16,47 +16,48 @@
 
 package com.google.android.gnd.persistence.sync;
 
-import androidx.work.Constraints;
 import androidx.work.NetworkType;
 import androidx.work.OneTimeWorkRequest;
 import androidx.work.WorkManager;
+import com.google.android.gnd.persistence.local.LocalValueStore;
 import io.reactivex.Completable;
 import javax.inject.Inject;
 import javax.inject.Provider;
 
 /** Enqueues file download work to be done in the background. */
-public class TileDownloadWorkManager {
-  private static final Constraints CONSTRAINTS =
-      new Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED).build();
+public class TileDownloadWorkManager extends BaseWorkManager {
 
-  private final Provider<WorkManager> workManagerProvider;
+  private final LocalValueStore localValueStore;
 
   @Inject
-  public TileDownloadWorkManager(Provider<WorkManager> workManagerProvider) {
-    this.workManagerProvider = workManagerProvider;
+  public TileDownloadWorkManager(
+      Provider<WorkManager> workManagerProvider, LocalValueStore localValueStore) {
+    super(workManagerProvider);
+    this.localValueStore = localValueStore;
+  }
+
+  @Override
+  Class<TileDownloadWorker> getWorkerClass() {
+    return TileDownloadWorker.class;
+  }
+
+  @Override
+  protected NetworkType preferredNetworkType() {
+    return localValueStore.shouldDownloadOfflineAreasOverUnmeteredConnectionOnly()
+        ? NetworkType.UNMETERED
+        : NetworkType.CONNECTED;
   }
 
   /**
    * Enqueues a worker that downloads files when a network connection is available, returning a
-   * completeable upon enqueueing.
+   * completable upon enqueueing.
    */
   public Completable enqueueTileDownloadWorker() {
     return Completable.fromRunnable(this::enqueueTileDownloadWorkerInternal);
   }
 
   private void enqueueTileDownloadWorkerInternal() {
-    OneTimeWorkRequest request = buildWorkerRequest();
-
+    OneTimeWorkRequest request = buildWorkerRequest(null);
```

Comment:
Consider adding a noargs override of `buildWorkerRquest()` instead of passing null here with JavaDoc explaining the default behavior with noargs.